### ISSUE TITLE
[BUG]: Nightly controller misdetection issue

### ISSUE THREAD

**Issue by DonKatsu** [PCSX2/pcsx2#11581]
### Describe the Bug

With the 2.0.2 appimage, my 8BitDo Pro 2 is detected as `SDL-0: Xbox One Controller` and all inputs work as expected.
With the 2.1.9 Nightly appimage, it is detected as `SDL-0: 8BitDo Zero 2`. This seems to change what the inputs are reported as.
Going from 2.0.2 to 2.1.9, some of the registered controller settings were not working in games.

Trying to set them again showed several problems/differences:
The DPad will not register
The left analog reports as `SDL-0/DPad[Direction]
The right analog registers as SDL-0/-+Axis9/10 instead of SDL-0/-+RightX/Y
L2/R2 register as SDL-0/Axis8/11 instead of SDL-0/[Side]Trigger
L3/R3 register as SDL-0/Button29/30 instead of SDL-0/[Side]Stick

The DPad values are still the same between version, but it does not work in games.
The analog sticks still work on 2.1.9 with the 2.0.2 values.

[2.0.2 screenshot](https://github.com/user-attachments/assets/658f5972-e8b2-4114-aa95-4d5b7723a38d)
[2.1.9 screenshot](https://github.com/user-attachments/assets/8975e217-b229-4588-b057-00fb880892ec)

### Reproduction Steps

1) Launch 2.0.2 and set Controller Settings
2) Launch 2.1.9 and set Controller settings

### PCSX2 Revision

v2.1.9

### Operating System

Linux (64bit) - Specify distro below

### If Linux - Specify Distro

CachyOS

**Comment by TheTechnician27** [comment]
That's strange. The PAD database was just updated in line with when you began experiencing the problem, but it only updated an Xbox One S controller scheme. https://github.com/PCSX2/pcsx2/commit/c0dce9f64bb936836e9c4bada351298e6ec1f9e8 I'll check what other issues may have caused this.

**Comment by TheTechnician27** [comment]
This seems like an upstream issue, so I'm reporting it as a bug there now.

**Comment by DonKatsu** [comment]
I tried some more things and observed the following:
I've been running my 8BitDo Pro 2 with the mode switch set to "X" for xinput since I got it. Currently that shows in Plasma 6 System Settings>Game Controllers as `Xbox One S Controller (/dev/input/event27)` over bluetooth. `Xbox 360 Controller (/dev/input/event27)` over usb.
Repairing with the mode switch set to "D" for dinput, it shows there as `8BitDo Pro 2 (/dev/input/event27)`. `8BitDo Pro 2 (/dev/input/event28)` over usb.

With it set to dinput, pcsx2 correctly detects it as an 8BitDo Pro 2 over bluetooth and usb. All inputs match that previous 2.0.2 screenshot and work after performing Automatic Mapping, or manually mapping.
The only thing unusual is that manually mapping the left and right triggers applies `SDL-0 Button29/30 & SDL-0/[Directon]Trigger` instead of just the trigger like Automatic Mapping does. Those mappings still seem to work as expected, however.

On another PC running Fedora 39 KDE (Plasma 5.27.11), my 8BitDo Pro 2 in xinput mode is detected as `8BitDo Pro 2 (/dev/input/js0)` in System Settings>Game Controllers over bluetooth. I'm not sure what changed there between that environment and this CachyOS one. But it still shows in pcsx2 2.1.9 Nightly as a Zero 2 with xinput.

Plasma's bluetooth applet on both PCs shows it as 8BitDo Pro 2 regardless of xinput or dinput.

dinput seems to work fine, but I can't really find a definitive answer on which mode I should be using on Linux.
I'm assuming the IDs in the PAD database are for dinput mode? Perhaps user error on my part then, but then adding the xinput IDs couldn't hurt?

**Comment by RedPanda4552** [comment]
We don't maintain the database, that is a SDL thing which we consume to assist with automatic mapping. That said, if they bunged it up it would definitely be worth reporting upstream.

The confusing part is yeah, why would Xinput mode be affected? Because it is relatively "stiff" in its mapping layout compared to Dinput... Just to confirm, when you use the Xinput mode, are you using the Automatic Mapping button? Or are you trying to map it manually?

**Comment by DonKatsu** [comment]
Will add on to the issue someone else made over on the SDL_GameControllerDB.
> Just to confirm, when you use the Xinput mode, are you using the Automatic Mapping button?

Right,. so automatic Mapping in xinput mode does apply the working values, and it's manually mapping applies the wrong ones.

**Comment by DonKatsu** [comment]
Well that didn't really go anywhere. But it did get me to look at what Steam had for the SDL output, and from what I can tell my Pro 2 in xinput still perfectly matches GUIDs and input profiles for the relevant Xbox controllers, so I suppose it isn't the DB's fault?
[steam sdl.txt](https://github.com/user-attachments/files/16318976/steam.sdl.txt)
But I really don't understand how just adding one new entry to the DB would throw off detection in pcsx2.
Ultimately, I'm fine with running it on dinput mode as that does look to be the way to go on Linux. But this is still technically a problem...

**Comment by fkryvyts** [comment]
@DonKatsu 
Hello, I  have the same exact controller and same issue. You can actually keep using xinput, it is fairly easy fix - just copy modified game_controller_db.txt into your data folder (for example /home/<your user>/.var/app/net.pcsx2.PCSX2/config/PCSX2 if you are using flatpak)
As for what to modify in that file - either remove 050000005e040000e002000030110000  entry that corresponds to 8BitDo Zero 2 or 050082795e040000e002000003090000 - for Xbox One S 
Funnily enough, even with completely empty game_controller_db.txt 8BitDo Pro 2 controller is properly registered but not when those 2 above mentioned entries exist in  game_controller_db.txt
Not sure why it is happening though,  8BitDo Pro 2 and 8BitDo Zero 2 have entirely different ids

**Comment by Zekromaster** [comment]
Can confirm the issue still exists. Found out that it was already reported for PCSX2 when it was mentioned in a PR I made to the Game Controller DB I made after encountering the issue in Duckstation. I too believed it was upstream, but the same discoveries made here were pointed out to me.

8BitDo Pro 2 and 8BitDo Zero 2 seem to have identical GUIDs up until the 24th character, could it be that somehow we're operating on truncated GUIDs?
